File: Easy/Sorted Squared Array/Sorted Squared Array.cpp
#include <vector>
#include <math.h>
using namespace std;

vector<int> sortedSquaredArray(vector<int> array)
{
    // Write your code here.
    int index = array.size() - 1;
    vector<int> output(index + 1);
    int small = 0, large = index;
    while (small <= large)
    {
        if (abs(array[small]) >= abs(array[large]))
        {
            output[index] = array[small] * array[small];
            small++;
        }
        else
        {
            output[index] = array[large] * array[large];
            large--;
        }
        index--;
    }
    return output;
}

File: Easy/Sorted Squared Array/Sorted Squared Array.py
def sortedSquaredArray(array):
    left, right = 0, len(array)-1
    index = right
    output = [0] * len(array)
    while(left <= right):
        if array[left]**2 > array[right]**2:
            output[index] = array[left]**2
            left += 1
        else:
            output[index] = array[right]**2
            right -= 1
        index -= 1
    return output
